test/distributed/_tensor/test_dtensor_compile.py

```diff
@@ -346,9 +348,11 @@ def fn(x, y):
             dt2 = DTensor.from_local(y.reshape(4, 2), mesh, [Shard(1)], run_check=False)
             dt_out = torch.matmul(dt, dt2)
             dt_out_redistribute = dt_out.redistribute(mesh, [Replicate()])
-            return dt_out.to_local()
+            return dt_out_redistribute.to_local()
```

Comment:
cc: @wanchaol I had to change this to pass this unit test `test_compile_dtensor_redistribute_backward`.

Reply:
that's interesting, but that seems a bugfix to me haha